app/controllers/api/account.php

```diff
@@ -2019,6 +2049,27 @@
                 ->setAttribute('passwordUpdate', DateTime::now());
         }
 
+        $target = $dbForProject->findOne('targets', [
+            Query::equal('identifier', [$email]),
+        ]);
+
+        if ($target && !$target->isEmpty()) {
+            throw new Exception(Exception::USER_TARGET_ALREADY_EXISTS);
+        }
+
+        /**
+        * @var Document $oldTarget
+        */
+        $oldTarget = $user->find('identifier', $oldEmail, 'targets');
+
+        if ($oldTarget !== false && !$oldTarget->isEmpty()) {
+            try {
+                $dbForProject->updateDocument('targets', $oldTarget->getId(), $oldTarget->setAttribute('identifier', $email));
+            } catch (Duplicate) {
```

Comment:
Do we need the try/catch? I think duplicate should only be thrown on create

Reply:
removed this